plugins/metrics/src/main/java/org/apache/cloudstack/metrics/MetricsServiceImpl.java

```diff
@@ -140,7 +145,9 @@ public InfrastructureResponse listInfrastructure() {
         response.setStoragePools(storagePoolDao.listAll().size());
         response.setImageStores(imageStoreDao.listImageStores().size());
         response.setSystemvms(vmInstanceDao.listByTypes(VirtualMachine.Type.ConsoleProxy, VirtualMachine.Type.SecondaryStorageVm).size());
-        response.setRouters(domainRouterDao.listAll().size());
+        response.setRouters(domainRouterDao.listByRole(VirtualRouter.Role.VIRTUAL_ROUTER).size());
```

Comment:
@Pearl1594 calling listAll() i.e. getting all the rows will make the APIs slow in a large env; this is fine but can you explore a way to simply get the count (instead of the rows).